specs/required-inputs/README.md

```diff
@@ -85,6 +85,8 @@ We will file a Chromium bug for the issue affecting the `nimble-combobox` and `n
 
 The `required` attribute only plays a role in validation for template-driven forms (with reactive forms, it is only used for accessibility purposes). Nimble inputs already support setting the `required` attribute via template, but we will add it to the directive wrappers for completeness.
 
+Note also that Angular's form validation support [automatically disables the native HTML form validation](https://v17.angular.io/api/forms/NgForm#native-dom-validation-ui) that would otherwise be triggered by setting the `required` attribute. This means that our Angular clients are not exposed to any of the native HTML validation bugs listed earlier in this document.
```

Comment:
I'm trying to understand the reason to mix the patterns. i.e. we don't style against the `:valid` pseudoselectors for error state visuals but we do key off the required state for the required visual. Have any reasons that we shouldn't follow the same pattern (i.e. have a separate `show-required` attribute like we have `show-error`)?

Reply:
- `:valid`/`:invalid` are based on native HTML validation. Since Angular validation disables this validation with the `novalidate` attribute, our error styling wouldn't work in Angular if it were based on `:invalid`.
- Despite not using native validation, Angular apps still use the `required` attribute to mark an input required. For template-driven forms, this causes the necessary validation to occur, but even for reactive forms, where validator functions are specified in code, it is expected that you still apply the `required` attribute in the template, for accessibility purposes (see the last sentence under [this heading](https://v17.angular.io/guide/form-validation#validating-input-in-reactive-forms)).
- Since both vanilla HTML and Angular clients already have to add the `required` attribute to their required input elements, it seems convenient and intuitive for our "required" visual (i.e. asterisk) to key off of that. I do not see any benefit to decoupling from that, especially since, from a consistency standpoint, all required inputs in Nimble-styled apps should use our visual affordance. It seems redundant and more error-prone to expect clients to set both `require` and `required-visible` attributes.

Reply:
Discussed more offline about the consistency of decoupling the visuals from the validation states which are currently reflected in the HLD.